pkg/services/store/object/http.go

```diff
@@ -0,0 +1,165 @@
+package object
+
+import (
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/middleware"
+	"github.com/grafana/grafana/pkg/web"
+
+	"github.com/grafana/grafana/pkg/api/response"
+	"github.com/grafana/grafana/pkg/api/routing"
+	"github.com/grafana/grafana/pkg/models"
+)
+
+type HTTPObjectStore interface {
+	// Register HTTP Access to the store
+	RegisterHTTPRoutes(routing.RouteRegister)
+}
+
+type httpObjectStore struct {
+	store ObjectStoreServer
+	log   log.Logger
+}
+
+func ProvideHTTPObjectStore(store ObjectStoreServer) HTTPObjectStore {
+	return &httpObjectStore{
+		store: store,
+		log:   log.New("http-object-store"),
+	}
+}
+
+// All registered under "api/object"
+func (s *httpObjectStore) RegisterHTTPRoutes(route routing.RouteRegister) {
+	// For now, require admin for everything
+	reqGrafanaAdmin := middleware.ReqSignedIn //.ReqGrafanaAdmin
+
+	// Every * must parse to a GRN (uid+kind)
+	route.Get("/store/*", reqGrafanaAdmin, routing.Wrap(s.doGetOject))
+	route.Get("/raw/*", reqGrafanaAdmin, routing.Wrap(s.doGetRawOject))
+	route.Post("/store/*", reqGrafanaAdmin, routing.Wrap(s.doWriteObject))
+	route.Delete("/store/*", reqGrafanaAdmin, routing.Wrap(s.doDeleteObject))
+	route.Get("/history/*", reqGrafanaAdmin, routing.Wrap(s.doGetHistory))
+	route.Get("/list/*", reqGrafanaAdmin, routing.Wrap(s.doListFolder)) // Simplified version of search -- path is prefix
+	route.Get("/search", reqGrafanaAdmin, routing.Wrap(s.doSearch))
+}
+
+// GRN? path >> UID + kind???
+func parseRequestParams(req *http.Request) (uid string, kind string, params map[string]string) {
```

Comment:
this approach to params seems a little flakey / strange to me. why use `.` as the delimiter and mix uid with kind vs more explicit `?uid=x` query params format?

Or why include kind at all vs looking object up just by it's `uid` (at which point we can infer kind maybe?)

Reply:
yes! exactly :)

I added a comment that we should address this ASAP.  In general this comes down to how do we best encode uid+kind.  It could be path+suffix, or may be more like key value pairs.